fe/fe-core/src/main/java/com/starrocks/sql/optimizer/Utils.java

```diff
@@ -607,21 +608,25 @@ public static int computeMinGEPower2(int num) {
     }
 
     public static boolean canEliminateNull(Set<ColumnRefOperator> nullOutputColumnOps, ScalarOperator expression) {
-        Map<ColumnRefOperator, ScalarOperator> m = nullOutputColumnOps.stream()
-                .map(op -> new ColumnRefOperator(op.getId(), op.getType(), op.getName(), true))
-                .collect(Collectors.toMap(identity(), col -> ConstantOperator.createNull(col.getType())));
-
-        for (ScalarOperator e : Utils.extractConjuncts(expression)) {
-            ScalarOperator nullEval = new ReplaceColumnRefRewriter(m).rewrite(e);
-
-            ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
-            // Call the ScalarOperatorRewriter function to perform constant folding
-            nullEval = scalarRewriter.rewrite(nullEval, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
-            if (nullEval.isConstantRef() && ((ConstantOperator) nullEval).isNull()) {
-                return true;
-            } else if (nullEval.equals(ConstantOperator.createBoolean(false))) {
-                return true;
+        try {
+            Map<ColumnRefOperator, ScalarOperator> m = nullOutputColumnOps.stream()
+                    .map(op -> new ColumnRefOperator(op.getId(), op.getType(), op.getName(), true))
+                    .collect(Collectors.toMap(identity(), col -> ConstantOperator.createNull(col.getType())));
+            for (ScalarOperator e : Utils.extractConjuncts(expression)) {
+                ScalarOperator nullEval = new ReplaceColumnRefRewriter(m).rewrite(e);
+
+                ScalarOperatorRewriter scalarRewriter = new ScalarOperatorRewriter();
+                // Call the ScalarOperatorRewriter function to perform constant folding
+                nullEval = scalarRewriter.rewrite(nullEval, ScalarOperatorRewriter.DEFAULT_REWRITE_RULES);
+                if (nullEval.isConstantRef() && ((ConstantOperator) nullEval).isNull()) {
+                    return true;
+                } else if (nullEval.equals(ConstantOperator.createBoolean(false))) {
+                    return true;
+                }
```

Comment:
constant true also can be eliminated, suggest to use and concatenate the remaining conjuncts.

Reply:
Add some comments. Only false or null can eliminate null.